testkit/steppedtask.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testkit
+
+import (
+	"testing"
+	"time"
+
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/util/taskstop"
+	"github.com/stretchr/testify/require"
+)
+
+// SteppedTaskFunc is the func for the task
+type SteppedTaskFunc func(ch *taskstop.Chan)
+
+// SteppedTask is a task object that can be paused
+type SteppedTask struct {
```

Comment:
This design is over complex to me that I feel it is too fine-grained. Task, Command, Runner, TestKit.... Can we somehow simplify it, leaving kit and command/task exposed to the user?